### ISSUE TITLE
the path to a 0.7.0 release

### ISSUE THREAD

**Issue by kdere** [chianti-atomic/ChiantiPy#82]
I have cleared up the problem with ionization cross-sections but I now need to check the ionization rates.  This is the main problem that 0.7.0 is solving and the only outstanding one that I am aware of.

The documentation is improved but certainly not perfect but that can be worked on more after the release.

I want to have a compressed tarball available on github and PyPI.  The sourceforge and chiantidatabase sites will then point to these two.

Anything else?

**Comment by kdere** [comment]
I am pretty confidant that the ionization cross-sections and rates agree with those from IDL.  The original calculations were done in IDL.

**Comment by wtbarnes** [comment]
Sorry I just now saw this. I've been a bit out of the loop with Chianti lately.

The GitHub repo can already be downloaded as a zip file. If possible, I think it best to first port all of the history of the sourceforge repo to GitHub and then delete the sourceforge page entirely. I think hosting the code multiple places mostly confuses people and versions on one site or the other fall out of date quickly.

Agree about the PyPI release.

Another possibility for distribution is putting ChiantiPy on the openastronomy conda channel. This makes the install really easy for those who use the Anaconda distribution. @Cadair has expressed interest in helping with this I think.

**Comment by wtbarnes** [comment]
I think it would be nice to clean up and update some of the tutorials and examples before doing a new release as well.

**Comment by Cadair** [comment]
I am happy to make a conda package for chiantipy, as this would be needed for SunPy to rely on ChiantiPy in the future. I had previously voiced the idea of making a separate package for the database do you think this is worthwhile?

**Comment by kdere** [comment]
putting ChiantiPy on Anaconda is fine with me.  Also, we release the database as a gzipped tar file and at that point it is free for anyone to do anything with it.  So, putting it also on conda, from the released tarball is also fine.

**Comment by kdere** [comment]
We could save all of the stuff that is under svn on sourceforge but this is not my first priority.  What I would like to do is a 0.7.0 release as the major error in the ionization cross-sections and rates has been fixed and verified.  
On github, I guess that means creating a tag and making a release.  I see that the SunPy releases include various tars etc of the package so we should do the same.
Also put it on PyPI.  I would not but it on Sourceforge except a simple statement that ChiantiPy has moved to Github.
I asked that the ChiantiPy ASCL.net page be updated to show github but that has not happened yet.  I did that some time ago.
Finally, send out a message on the chianti google group

**Comment by wtbarnes** [comment]
Re: moving the svn history from sourceforge, yes I did not mean to imply that should be done before the release (it may be a bit of a task!)

Ah yes I see what you mean about the tarballs as part of the releases. I had forgotten about that. 

I also sent a request to ASCL and noticed that the entry had not been updated. I think they are a bit slow when it comes to updating some of their references.

Re: the versioning, since this release is primarily a bug fix, maybe it should be 0.6.6 instead of 0.7? Then release 0.7 once the documentation and tutorials/examples have been a bit more revised. This isn't really that important except that incrementing the second digit in version numbers is typically indicative of major changes in the code.

**Comment by kdere** [comment]
I have been updating the documentation and I would like to do a 0.7.0 release soon.

One problem with the documentation is that sphinx is expecting to find the method setup but it is now _setup so it does not get documented.

**Comment by wtbarnes** [comment]
This is in reference to the [`_setup` method on the `ion object`?](https://github.com/chianti-atomic/ChiantiPy/blob/master/ChiantiPy/core/Ion.py#L183)

Methods with an underscore are considered to not be part of the public API and so are not documented. I guess since the user has the option for this method to *not* be called in the constructor, then it should be public so both `_setup` and `_setupIonrec` should have the underscores removed.

**Comment by kdere** [comment]
I agree.  I will do that

**Comment by kdere** [comment]
just removing the under-score broke something, will check it out

**Comment by wtbarnes** [comment]
Where specifically is it failing? Could you paste the Traceback?

**Comment by kdere** [comment]
it does not crash, just doesn't work like it should.
in $HOME/chianti/sutherland
there is a k_4.elvlc, .wgfa and .scups
 > k4 = ch.ion('k -4',1.e+5, 1.e+9, setup=0)
>k4.setup(alternate_dir='/home/ken/chianti/sutherland')
and I get
elvlc file does not exist:  /data2/svn/chianti/dbase/trunk/k/k_4/k_4.elvlc

it should be looking in $HOME/chianti/sutherland

the setup method with the alternate directory is really for code/data developers but it is very handy for working with data that is not yet in the CHIANTI repository

**Comment by wtbarnes** [comment]
Ok. Can you create a new issue for this?

**Comment by kdere** [comment]
PyPI wants a README.txt file that is in rst format so it can be properly displayed.  I will see if I can dig up an old one and go from there

**Comment by wtbarnes** [comment]
see #97. This is a direct translation from the markdown version

**Comment by kdere** [comment]
just what we needed.  I plan to do a release this week, hopefully sooner than later

**Comment by kdere** [comment]
Is astropy really a dependency?  I think we need to list IPython(3) and Jupyter(3) as very helpful but not dependencies

**Comment by kdere** [comment]
I just made a 0.7.0 release.  I did not do anything about making a tag.  I will put it on PyPi soon

**Comment by wtbarnes** [comment]
0.7 has been released. This issue can be closed.
